src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs

```diff
@@ -51,6 +55,9 @@ internal static class ITextBufferVisibilityTrackerExtensions
             if (service is null)
                 return;
 
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
             await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
❗ Cancellation here (during the switch) will result in an exception

Reply:
good call.

Reply:
I've been discussing with Andrew a new `GetResult(out Exception)` method that would support this case and also support the feature from #67487.